### DISCUSSION TITLE
Invalid Message ID

### DISCUSSION THREAD

**Discussion by sairmali** [python-smpplib/python-smpplib#240] [Q&A]
import smpplib.client
import smpplib.consts

# Initialize your SMPP client
client = smpplib.client.Client('127.0.01', 2775, allow_unknown_opt_params=True)
client.connect()

try:
    # Bind to the SMPP server with credentials
    client.bind_transceiver(system_id='sarimali', password='password')
    # Query the delivery status for a specific message ID
    message_id_to_check = '2C16E7kk'
    # Sending the query request
    status = client.query_message(
        message_id=message_id_to_check,
        source_addr_ton=5,
        source_addr_npi=0,
        source_addr='Dar'
    )
    # Interpret the status and print the result
    client.listen()
    if status.status == 0:
        delivery_status = "Delivered" if status.status else "Not Delivered"
        print(f"Delivery Status: {delivery_status}")
    else:
        print("Query Failed")
except smpplib.exceptions.PDUError as e:
    print(f"Error querying delivery status: {e}")
finally:
    client.disconnect()

-----------------Response-------------------------

<smpplib.client.Client object at 0x000001D7D08520D0> is disconnecting in the bound state
Error querying delivery status: ('(12) query_sm_resp: Invalid Message ID', 12)

**Comment by eigenein** [comment]
Hi, I don't quite understand the question, could you elaborate?

**Comment by sairmali** [comment]
hello @eigenein ,I'm currently running a script to fetch the delivery status of a message through its unique message ID using the (query_message) command. However, my script encounters an error during this process.

**Reply by eigenein** [reply]
Sure, I can see that, but the error is returned by your SMSC provider

**Reply by sairmali** [reply]
1. ![image](https://github.com/python-smpplib/python-smpplib/assets/54973008/ac4edee8-e3e4-4942-9f86-9420ccc26e39)
this response given from my script side by message id


2. ![WhatsApp Image 2023-12-10 at 19 46 33_bd57f37e](https://github.com/python-smpplib/python-smpplib/assets/54973008/cf031a44-b3c4-4f3d-b90a-ec7bb20d888b)

this mail my provider send he say this problem form script side

**Reply by sairmali** [reply]
this is mail body send by my provider

"Kindly note that below message ID valid and the SMS delivered successfully to the destination based on our SMSC logs below , So the issue on your SMPP software/conversion ."

**Comment by sairmali** [comment]
@eigenein please tell me ?

**Comment by physicsb0y** [comment]
i am also getting the the same error, Invalid message id, has anyone found the solution for this?
